lib/smack/SmackRep.cpp

```diff
@@ -76,8 +137,20 @@ string SmackRep::type(const llvm::Type* t) {
     return BOOL_TYPE;
   else if (isFloat(t))
     return FLOAT_TYPE;
-  else
+  else if (isInt(t)) {
+    if (BIT_VECTOR) {
+      stringstream s;
+      s << "i" << getIntSize(t);
+      return s.str();
+    } else
+      return getPtrType();
+  } else if (isPointer(t))
     return getPtrType();
+  else if (t->isAggregateType())
+    return getPtrType(); 
```

Comment:
How about we cutting off else if (isInt(t) || isPointer(t) || t->isAggregateType()) and unsupported type branches since SMACK never encounters unsupported types?

Then the code is,
else if (isInt(t) && BIT_VECTOR)
  return int_type(getIntSize(t));
else
  return getPtrType();

If we do in this way else if (isInt(t) && BIT_VECTOR)
  return int_type(getIntSize(t));
else if (isInt(t) || isPointer(t) || t->isAggregateType()) {
  return getPtrType();

isInt(t) may be evaluated twice before SmackRep::type() returns getPtrType() for a non-bit-vector int if the compiler does not perform any optimization.